### DISCUSSION TITLE
Intro timing with non-osu! theme can be better

### DISCUSSION THREAD

**Discussion by NotGumballer91** [ppy/osu#20662] [Ideas] [answered]
Currently, Lazer starts immediately on the song's timing point, fading in.

Although it's better than before, It can be better.
First, It's not really *synced* as the first one. I admit it's quite abrupt and sudden.

**How about a bit of both? My solution would be:**
Fade **into the timing point.** So it can fade into the drop as the osu cookie shows up.

I'll illustrate later if it's confusing.

**Comment by nagi-desuuu** [comment]
This was actually the "prototype" version.
Check out https://github.com/ppy/osu/discussions/19687#discussioncomment-3373037

**Comment by peppy** [comment]
Feel free to submit a PR improving further. Make sure to test against a wide variety of beatmaps.

**Reply by NotGumballer91** [reply]
~~Would you mind sending stable's code for intro-timing?~~

**Comment by NotGumballer91** [comment, marked as answer]
#20687
